systemtest/src/test/java/io/strimzi/systemtest/migration/MigrationST.java

```diff
@@ -0,0 +1,265 @@
+import io.strimzi.systemtest.utils.kubeUtils.objects.PodUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import java.util.Map;
+
+import static io.strimzi.systemtest.TestConstants.REGRESSION;
+import static io.strimzi.systemtest.resources.ResourceManager.cmdKubeClient;
+import static io.strimzi.systemtest.resources.ResourceManager.kubeClient;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+@Tag(REGRESSION)
+public class MigrationST extends AbstractST {
+
+    private static final Logger LOGGER = LogManager.getLogger(MigrationST.class);
+    private static final String CONTINUOUS_SUFFIX = "-continuous";
+
+    /**
+     * @description This testcase is focused on migration process from ZK to KRaft.
+     * It goes through whole process, together with checking that message transmission throughout the test will not be
+     * disrupted.
+     *
+     * @steps
+     *  1. - Deploys Kafka resource (with enabled NodePools and KRaft set to disabled) with Broker NodePool
+     *  2. - Deploys Controller NodePool - it's created here, so we will firstly delete KafkaTopics in our ResourceManager;
+     *       the Pods will not be created until the migration starts
+     *  3. - Creates topics for continuous and immediate message transmission, TLS user
+     *  4. - Starts continuous producer & consumer
+     *  5. - Does immediate message transmission
+     *  6. - Starts the migration
+     *  7. - Annotating the Kafka resource with strimzi.io/kraft:migration
+     *  8. - Controllers will be created and moved to RUNNING state
+     *  9. - Two rolling updates of the Broker Pods
+     *  10. - Checks that Kafka CR has .status.kafkaMetadataState set to KRaftPostMigration
+     *  11. - Creates a new KafkaTopic and checks both ZK and KRaft metadata for presence of the KafkaTopic
+     *  12. - Does immediate message transmission to the new KafkaTopic
+     *  13. - Finishes the migration - annotates the Kafka resource with strimzi.io/kraft:enabled
+     *  14. - Controller Pods will be rolled
+     *  15. - Checks that Kafka CR has .status.kafkaMetadataState set to KRaft
+     *  16. - Removes LMFV and IBPV from Kafka configuration
+     *  17. - Broker and Controller Pods will be rolled
+     *  18. - Creates a new KafkaTopic and checks KRaft metadata for presence of the KafkaTopic
+     *  19. - Does immediate message transmission to the new KafkaTopic
+     *  20. - Waits until continuous clients are finished successfully
+     *
+     * @usecase
+     *  - zk-to-kraft-migration
+     */
+    @IsolatedTest
+    @SuppressWarnings("checkstyle:MethodLength")
+    void testMigrationFromZkToKRaft(ExtensionContext extensionContext) {
+        TestStorage testStorage = new TestStorage(extensionContext, TestConstants.CO_NAMESPACE);
```

Comment:
i guess that now all resources with end up in `co-namespace` and not  `test-suite-namespace` ?  is this desired ?

Reply:
This will not change the way, how it is already implemented. By default, everything is installed in `co-namespace` in case of `@IsolatedTest`. But I don't have an answer why it's not installed in `test-suite-namespace` 🤷‍♂️ 
The `TestStorage` will hold there just the value, so the test checks will look into the right namespace.

This behavior can be investigated as part of a different PR